addons/account/models/chart_template.py

```diff
@@ -502,9 +502,13 @@ def deref_values(values, model):
                 ):
                     try:
                         values[fname] = self.ref(value).id if value not in ('', 'False', 'None') else False
-                    except ValueError as e:
-                        _logger.warning("Failed when trying to recover %s for field=%s", value, field)
-                        raise e
+                    except ValueError:
+                        # We can't find the record referenced in the chart template in our database.
+                        # This might happen when we're creating a branch and the parent company has deleted the
+                        # referenced record and replaced it with something else.
+                        #
+                        # In this case, we try looking for the record already set on the company or its root.
+                        values[fname] = self.env.company[fname] or self.env.company.parent_ids[0][fname] or False
```

Comment:
won't that fail in the same way if the company we're creating and trying to setup isn't a child company, but just another company without any specific relationship but being in the same country? 
:thinking:

Reply:
and where is it said that model is 'res.company' ? from the docstring, it looks like we could be trying to get values from anything, actually... :suspect:

Reply:
> won't that fail in the same way if the company we're creating and trying to setup isn't a child company, but just another company without any specific relationship but being in the same country?
> 🤔

No, because in that case all chart template data is loaded before, their xmlid's prefixed with the new company's id. Otherwise we would already have lots of issues because it previously just threw an error.

> and where is it said that model is 'res.company' ? from the docstring, it looks like we could be trying to get values from anything, actually... :suspect:

You're right actually. I added a check before to see if the model is `res.company` or not.